lighthouse-core/config/v2/config.js

```diff
@@ -0,0 +1,182 @@
+const path = require('path');
+
+const defaultConfigPath = './default.json';
+const defaultConfigJson = require(defaultConfigPath);
+
+const _flatten = arr => [].concat(...arr);
+const _subtract = (setA, setB) => Array.from(setA).filter(x => !setB.has(x));
+
+class ConfigV2 {
+  constructor(configJson, configPath) {
+    if (!configJson) {
+      configJson = defaultConfigJson;
+      configPath = path.resolve(__dirname, defaultConfigPath);
+    }
+
+    // Perform a shallow clone so we can adjust gatherers and audits
+    configJson = Object.assign({}, configJson);
```

Comment:
> or at least I try to do shallow copies everywhere, we'll see when I add tests :)

test will be good :) `assert.notDeepStrictEqual` might be sufficient, but we could also do something with deep `Object.freeze` and see what throws (in strict mode)

Especially important for the browserified/lighthouse-as-a-module case since when loading the config json via `require()` any changes to the original object is persisted to the next use of it.